include/asterisk/app.h

```diff
@@ -1196,9 +1196,54 @@ struct ast_group_info;
 /*! \brief Split a group string into group and category, returning a default category if none is provided. */
 int ast_app_group_split_group(const char *data, char *group, int group_max, char *category, int category_max);
 
+/* Create group meta if it doesn't exist  */
+int ast_app_group_add_meta_ifneeded(const char *group, const char *category);
+
+/* This removes the the group metadata for a group@category if there are no more channels in the group@category */
+int ast_app_group_remove_meta_ifneeded(const char *group, const char *category);
+
+/*! \brief Remove channel assignments for the specified group  */
+int ast_app_group_remove_all_channels(const char *group, const char *category);
+
+/*! \brief Rename a group@category while retaining all the channel memberships
+ * \retval -1 on Error
+ * \retval 0 On zero rename operations performed (nothing found to rename)
+ * \retval >=1 On 1 or more rename operations performed
+ */
+int ast_app_group_rename(const char *old_group, const char *old_category, const char *new_group, const char *new_category);
+
 /*! \brief Set the group for a channel, splitting the provided data into group and category, if specified. */
 int ast_app_group_set_channel(struct ast_channel *chan, const char *data);
 
+/*!
+ * \brief Set a group variable for a group@category
+ *
+ * \param chan      channel (if any) that is setting the group variable (can be NULL)
+ * \param group     group to set the variable on (cannot be null)
+ * \param category  category to set the variable on (cannot be null)
+ * \param name      name of the variable to set (cannot be null)
+ * \param value     value of the variable to set (cannot be null)
+ *
+ * \retval 0 On success
+ * \retval -1 On failure (group@category not found)
+ * \retval -2 On failure (input validation error)
+ */
+int ast_app_group_set_var(struct ast_channel *chan, const char *group, const char *category, const char *name, const char *value);
+
+/*!
+ * \brief Get a group variable for a group@category
+ *
+ * \param group     group to get the variable from
+ * \param category  category to get the variable from
+ * \param name      name of the variable to get
+ *
+ * \retval NOT NULL On success return  char* of variable contents
```

Comment:
This one could just be `\return Allocated string containing variable contents`

Reply:
Fixed